src/main/java/me/ryanhamshire/GPFlags/flags/FlagDef_InfiniteArrows.java

```diff
@@ -24,23 +23,28 @@
     public void onProjectileHit(ProjectileHitEvent event) {
         if (event.getEntityType() != EntityType.ARROW && event.getEntityType() != EntityType.SNOWBALL) return;
 
-        Projectile arrow = event.getEntity();
-
-        ProjectileSource source = arrow.getShooter();
-        if (source == null || !(source instanceof Player)) return;
+        ProjectileSource source = event.getEntity().getShooter();
+        if (!(source instanceof Player)) return;
 
         Player player = (Player) source;
         if (player.getGameMode() == GameMode.CREATIVE) return;
 
-        Flag flag = this.GetFlagInstanceAtLocation(arrow.getLocation(), player);
+        Flag flag = this.GetFlagInstanceAtLocation(event.getEntity().getLocation(), player);
         if (flag == null) return;
 
         PlayerInventory inventory = player.getInventory();
-        ItemMeta meta = inventory.getItemInMainHand().getItemMeta();
-        if (meta != null && meta.hasEnchant(Enchantment.ARROW_INFINITE)) return;
+
+        if (event.getEntityType() == EntityType.SNOWBALL) {
+            inventory.addItem(new ItemStack(Material.SNOWBALL));
+            return;
+        }
+
+        Arrow arrow = (Arrow) event.getEntity();
+        if (arrow.getPickupStatus() != AbstractArrow.PickupStatus.ALLOWED) return;
```

Comment:
It worries me blindly casting the entity to an arrow, are we sure its an arrow?
I can't remember if a trident calls the projectile hit event, but if it does, it would thru an error at this point.
Also this is not future safe as Minecraft may possible add other projectile types.
Not to mention snowballs and arrows are not the only projectiles, as I said already tridents, but also thinks like fireballs (while that may never happen, its always a possibility)